test/integration/server/helpers/helpers.go

```diff
@@ -47,22 +47,32 @@ func CreateRepo(ctx context.Context, gp gitprovider.Client, url string) (gitprov
 		return nil, ref, fmt.Errorf("error parsing url: %w", err)
 	}
 
+	defaultBranch := "main"
 	repo, _, err := gp.OrgRepositories().Reconcile(ctx, *ref, gitprovider.RepositoryInfo{
 		Description:   gitprovider.StringVar("Integration test repo"),
 		Visibility:    gitprovider.RepositoryVisibilityVar(gitprovider.RepositoryVisibilityPrivate),
-		DefaultBranch: gitprovider.StringVar("main"),
+		DefaultBranch: gitprovider.StringVar(defaultBranch),
 	}, &gitprovider.RepositoryCreateOptions{AutoInit: gitprovider.BoolVar(true)})
 
 	if err != nil {
 		return nil, nil, fmt.Errorf("could not reconcile org repo: %w", err)
 	}
 
-	err = utils.WaitUntil(bytes.NewBuffer([]byte{}), 3*time.Second, 5*time.Second, func() error {
-		_, err := gp.OrgRepositories().Get(ctx, *ref)
+	err = utils.WaitUntil(os.Stdout, 3*time.Second, 9*time.Second, func() error {
+		r, err := gp.OrgRepositories().Get(ctx, *ref)
 		if err != nil {
 			return err
 		}
 
+		commits, err := r.Commits().ListPage(ctx, defaultBranch, 1, 0)
```

Comment:
:question: Why are we waiting for commits to occur on an operation that creates a repo?

Reply:
Because we are pushing changes to the repo and we want the repo to be ready to receive commits. When we run `gp.OrgRepositories().Get(ctx, *ref)` it doesn't mean necessarily that the repo is ready to receive commits. You can see the error this fixes here https://github.com/weaveworks/weave-gitops/runs/4172766374?check_suite_focus=true#step:13:125.